libopenage/curve/element_wrapper.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+#include "time/time.h"
+
+namespace openage::curve {
+
+    template<typename T>
+    struct element_wrapper {
+        // Insertion time of the element
+        time::time_t _alive;
+        // Erase time of the element
+        // TODO: this has to be mutable because erase() will complain otherwise
+        mutable time::time_t _dead;
+        // Element value
+        T value;
+
+        element_wrapper(const time::time_t &time, const T &value) :
+            _alive{time},
+            _dead{time::TIME_MAX},
+            value{value} {}
+
+        element_wrapper(const T &value, const time::time_t &alive, const time::time_t &dead) :
+            _alive{alive},
+            _dead{dead},
+            value{value} {}
+
+        const time::time_t &alive() const {
+            return _alive;
+        }
+
+        const time::time_t &dead() const {
+            return _dead;
+        }
+
+        // TODO: this has to be const because erase() will complain otherwise
+        void set_dead(const time::time_t &time) const {
+            _dead = time;
+        }
+    };

```

Comment:
Now that this struct moved to a separate file, we should document all its functions and constructors.